roles/core/hosts_file/templates/hosts.j2

```diff
@@ -59,55 +58,65 @@
-{% set host_eqg = (macro_accelerated.node_equipment_group(host)|trim) %}
-{% set host_definition = main_host[host_mg]['children'][host_eqg]['hosts'][host] %}
-{% set host_nic = host_definition['network_interfaces'] %}
-{% set host_bmc = host_definition['bmc'] %}
-{% set host_icb_net = (management_networks_naming + (host_icb | replace(iceberg_naming,' ') | trim | string)) %}
-{% set host_node_main_network = (macro_network.node_main_network(host_nic,host_icb_net) | trim) %}
-{% set host_node_main_network_interface = (macro_network.node_main_network_interface(host_nic,host_node_main_network) | trim) %}
-{% set alias_list = [] %}
-{% if host_definition['global_alias'] is defined and not none %}
-  {% for alias in host_definition['global_alias'] %}{{ alias_list.append(alias) }}{% endfor %}
-{% endif %}
-{% if host_definition['alias'] is defined and not none and host_icb == j2_current_iceberg %}
-  {% for alias in host_definition['alias'] %}{{ alias_list.append(alias) }}{% endfor %}
+{% if direct_access is not defined %}
+{# -- Standard method. Rely on hostvars. -- #}
+
+  {% for host in range %}
+{{ write_host(host,hostvars[host],hostvars[host]['j2_node_main_network_interface'],hostvars[host]['j2_current_iceberg']) }}
+  {% endfor %}
+
+{% else %}
+{# -- Direct access method. Rely on direct_access content. -- #}
+
+  {% for mg, mg_vars in direct_access.items() %}
+    {% for eqg, eqg_vars in mg_vars.children.items() %}
+      {% for host, host_vars in eqg_vars.hosts.items() %}
+        {% if host in range %}
+          {# Get host iceberg, from groups #}
+          {% set host_icb = (macro_groups.node_current_iceberg(host)|trim) %}
+          {# Check if user forced main network. #}
+          {% if host_vars.j2_node_main_network is defined %}
+            {% set host_node_main_network = host_vars.j2_node_main_network %}
+            {% set host_node_main_network_interface = (macro_network.node_main_network_interface(host_vars.network_interfaces,host_node_main_network) | trim) %}
+          {% elif host_vars.j2_node_main_network_interface is defined %}
+            {% set host_node_main_network_interface = host_vars.j2_node_main_network_interface %}
+          {% else %}
+            {% set host_icb_net = (management_networks_naming + (host_icb | replace(iceberg_naming,' ') | trim | string)) %}
+            {% set host_node_main_network = (macro_network.node_main_network(host_vars.network_interfaces,host_icb_net) | trim) %}
+            {% set host_node_main_network_interface = (macro_network.node_main_network_interface(host_vars.network_interfaces,host_node_main_network) | trim) %}
+          {% endif %}
+{{ write_host(host,host_vars,host_node_main_network_interface,host_icb) }}
+        {% endif %}
+      {% endfor %}
+    {% endfor %}
+  {% endfor %}
+
 {% endif %}
-{{write_host_2(host,host_nic,host_node_main_network_interface,alias_list,host_bmc)}}
-{% endfor %}
-#}
+
 ## External hosts
```

Comment:
This line leave an empty "## external hosts" line in /etc/hosts file when there is no external hosts, can you move this under the if statement, so this line is present only when external_hosts var is defined ?